src/main/java/com/glencoesoftware/omero/zarr/ZarrPixelBuffer.java

```diff
@@ -326,31 +320,42 @@ public void close() throws IOException {
      * Implemented as specified by {@link PixelBuffer} I/F.
      * @see PixelBuffer#checkBounds(Integer, Integer, Integer, Integer, Integer)
      */
+    @Override
     public void checkBounds(Integer x, Integer y, Integer z, Integer c,
             Integer t)
             throws DimensionsOutOfBoundsException {
         if (x != null && (x > getSizeX() - 1 || x < 0)) {
             throw new DimensionsOutOfBoundsException("X '" + x
-                    + "' greater than sizeX '" + getSizeX() + "'.");
+                    + "' greater than sizeX '" + getSizeX() + "' or < '0'.");
         }
         if (y != null && (y > getSizeY() - 1 || y < 0)) {
             throw new DimensionsOutOfBoundsException("Y '" + y
-                    + "' greater than sizeY '" + getSizeY() + "'.");
+                    + "' greater than sizeY '" + getSizeY() + "' or < '0'.");
         }
 
         if (z != null && (z > getSizeZ() - 1 || z < 0)) {
             throw new DimensionsOutOfBoundsException("Z '" + z
-                    + "' greater than sizeZ '" + getSizeZ() + "'.");
+                    + "' greater than sizeZ '" + getSizeZ() + "' or < '0'.");
         }
 
         if (c != null && (c > getSizeC() - 1 || c < 0)) {
             throw new DimensionsOutOfBoundsException("C '" + c
-                    + "' greater than sizeC '" + getSizeC() + "'.");
+                    + "' greater than sizeC '" + getSizeC() + "' or < '0'.");
         }
 
         if (t != null && (t > getSizeT() - 1 || t < 0)) {
             throw new DimensionsOutOfBoundsException("T '" + t
-                    + "' greater than sizeT '" + getSizeT() + "'.");
+                    + "' greater than sizeT '" + getSizeT() + "' or < '0'.");
+        }
+    }
+
+    public void checkReadSize(int[] shape) {
+        long length = length(shape);
+        long maxLength = maxPlaneWidth * maxPlaneHeight;
```

Comment:
Would it be worth doing long multiplication here too? Or do we know for sure that `maxPlaneWidth * maxPlaneHeight` will always be less than `Integer.MAX_VALUE`?

Reply:
We don't know for sure but if you were to configure your server like that all sorts of other stuff would break spectacularly.